pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -346,7 +348,9 @@ func createConfigStore(client kubelib.Client, revision string, opts Options) (mo
 	workloadEntriesSchemas := collection.NewSchemasBuilder().
 		MustAdd(collections.IstioNetworkingV1Alpha3Workloadentries).
 		Build()
-	return crdclient.NewForSchemas(client, revision, opts.DomainSuffix, workloadEntriesSchemas)
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	return crdclient.NewForSchemas(ctx, client, revision, opts.DomainSuffix, workloadEntriesSchemas)
```

Comment:
If I am reading this right, if we hit 10s timeout we will never recover? Since createConfigStore error just results in a log?

Reply:
changed it to return an error there

`secretcontroller` will retry this and treat the retry as an `update` event